lib/src/DonutChart.js

```diff
@@ -6,218 +6,156 @@
+  const [toggleSelect, setToggleSelect] = useState(false);
+
+  useEffect(() => {
+    setSelected(data[0]);
+    setToggleSelect(false);
+  }, [data]);
+
+  const total = data.reduce((sum, currItem) => sum + currItem.value, 0);
+  const graphWidth = legend ? (width * (2 / 3)) : width;
+
+  const itemProps = (item, index) => {
+    const { isEmpty, label } = item;
+    const isSelected = selected.label === label;
+    const isToggled = isSelected && toggleSelect;
+
+    return {
+      classNames: `${item.className ? item.className : ''} ${isEmpty ? 'empty' : ''} ${isSelected ? 'selected' : ''} ${isToggled ? 'toggled' : ''}`.trim(),
+      fill: isEmpty ? emptyColor : colorFunction(colors, index),
+      opacity: isSelected && !toggleSelect ? 0.5 : 1,
+      stroke: isEmpty ? emptyColor : strokeColor,
+      clickHandlers: {
+        onClick: () => {
+          if (selected.label === label) {
+            const toggle = clickToggle ? !toggleSelect : false;
+            setSelected(item);
+            setToggleSelect(toggle);
+            onClick(item, toggle);
+          }
+        },
+        onMouseEnter: () => {
+          if (!toggleSelect) {
+            setSelected(item);
+            onMouseEnter(item);
+          }
         }
-    }
-
-    /* React render function */
-    render() {
-        const {
-            startAngle, width, height, formatValues, className, data, legend,
-            emptyColor, strokeColor, colors, colorFunction, innerRadius,
-            outerRadius, emptyOffset, selectedOffset, toggledOffset
-        } = this.props;
-
-        const arcsClassName = `${className}-arcs`;
-        const innerTextClassName = `${className}-innertext`;
-        const legendClassName = `${className}-legend`;
-
-        const checkData = data.length ? data : [{
-            label: '',
-            value: 100,
-            isEmpty: true
-        }];
-
-        const total = calculateTotal(checkData);
-        const twoThirds = 2 / 3;
-        const graphWidth = legend ? (width * twoThirds) : width;
-        const legendWidth = width - graphWidth;
+      }
+    };
+  };
+
+  return (
+    <DonutChartContext.Provider value={{
+      ...restProps,
+      className,
+      data,
+      graphWidth,
+      itemProps,
+      selected,
+      toggleSelect,
+      total,
+      width
+    }}>
+      <svg className={ className } style={{ height, width }} viewBox={`0 0 ${width} ${height}`}>
+        <Arcs />
+        <g className={ `${className}-innertext` }>
+          <text
+            className={ `${className}-innertext-label` }
+            x={ graphWidth / 2 }
+            y="45%"
+            textAnchor="middle">
+            { selected.label }
+          </text>
+          <text
+            className={ `${className}-innertext-value` }
+            x={ graphWidth / 2 }
+            y="60%"
+            textAnchor="middle">
+            { formatValues(selected.value, total) }
+          </text>
+        </g>
+        {legend && (
```

Comment:
Previously [the separate `Legend` component](https://github.com/vonbearshark/react-donut-chart/pull/25/files#diff-5b53eea52b75f31f888f2accf09765da536531a470fe42207d0286c00d76bd16)